internal/api/upload.go

```diff
@@ -0,0 +1,82 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"net/http"
+	"os"
+)
+
+func (c *ApiClient) Upload(ctx Context, reportUploadType string, uploadDate string, email string, file *os.File) error {
+	var body bytes.Buffer
+	var uploadDateTransformed *model.Date
+	var req *http.Request
+
+	if uploadDate != "" {
+		uploadDateFormatted := model.NewDate(uploadDate)
+		uploadDateTransformed = &uploadDateFormatted
+	}
+
+	err := json.NewEncoder(&body).Encode(model.Upload{
+		ReportUploadType: reportUploadType,
+		UploadDate:       uploadDateTransformed,
+		Email:            email,
+		File:             file,
+	})
+	if err != nil {
+		return err
+	}
+
+	if reportUploadType == "DebtChase" {
```

Comment:
also I'd be tempted to refactor this to a `switch` that sets `reqPath` or something as that is the only thing that changes, then construct the request (or extract as a function)

Reply:
done